src/m365/spo/commands/tenant/tenant-commandset-set.ts

```diff
@@ -0,0 +1,223 @@
+        webTemplate: typeof args.options.webTemplate !== 'undefined',
+        location: args.options.location
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-i, --id <id>'
+      },
+      {
+        option: '-t, --newTitle [newTitle]'
+      },
+      {
+        option: '-l, --listType [listType]',
+        autocomplete: SpoTenantCommandSetSetCommand.listTypes
+      },
+      {
+        option: '-i, --clientSideComponentId [clientSideComponentId]'
+      },
+      {
+        option: '-p, --clientSideComponentProperties [clientSideComponentProperties]'
+      },
+      {
+        option: '-w, --webTemplate [webTemplate]'
+      },
+      {
+        option: '--location [location]',
+        autocomplete: SpoTenantCommandSetSetCommand.locations
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (!args.options.newTitle &&
+          !args.options.listType &&
+          !args.options.clientSideComponentId &&
+          !args.options.clientSideComponentProperties &&
+          !args.options.webTemplate &&
+          !args.options.location) {
+          return 'Specify at least one property to update';
+        }
+
+        if (args.options.clientSideComponentId && !validation.isValidGuid(args.options.clientSideComponentId)) {
+          return `${args.options.clientSideComponentId} is not a valid GUID`;
+        }
+
+        if (args.options.listType && SpoTenantCommandSetSetCommand.listTypes.indexOf(args.options.listType) < 0) {
+          return `${args.options.listType} is not a valid list type. Allowed values are ${SpoTenantCommandSetSetCommand.listTypes.join(', ')}`;
+        }
+
+        if (args.options.location && SpoTenantCommandSetSetCommand.locations.indexOf(args.options.location) < 0) {
+          return `${args.options.location} is not a valid location. Allowed values are ${SpoTenantCommandSetSetCommand.locations.join(', ')}`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    try {
+      const appCatalogUrl = await spo.getTenantAppCatalogUrl(logger, this.debug);
+      if (!appCatalogUrl) {
+        throw 'No app catalog URL found';
+      }
+
+      await this.updateTenantWideExtension(appCatalogUrl, args.options, logger);
+    }
+    catch (err: any) {
+      this.handleRejectedODataJsonPromise(err);
+    }
+  }
+
+  public async getTenantCommandSet(logger: Logger, options: Options, requestUrl: string): Promise<number> {
+    if (this.verbose) {
+      logger.logToStderr(`Getting the tenant command set ${options.id}`);
+    }
+
+    const filter: string = `startswith(TenantWideExtensionLocation,'ClientSideExtension.ListViewCommandSet') and Id eq '${options.id}'`;
+
+    const reqOptions: CliRequestOptions = {
+      url: `${requestUrl}/items?$filter=${filter}`,
```

Comment:
I'd use `items(4)` here to get the item by id. We can verify if it's actually a commandSet later.